test/runtests.jl

```diff
@@ -2,192 +2,179 @@ using PrecompileTools
 using Test
 using Pkg
 using UUIDs
+using Base: specializations
 
 @testset "PrecompileTools.jl" begin
-    specializations(m::Method) = isdefined(Base, :specializations) ? Base.specializations(m) : m.specializations
-
     push!(LOAD_PATH, @__DIR__)
 
     using PC_A
-    if VERSION >= v"1.8"
-        # Check that calls inside @setup_workload are not precompiled
-        m = which(Tuple{typeof(Base.vect), Vararg{T}} where T)
-        have_mytype = false
-        for mi in specializations(m)
-            mi === nothing && continue
-            have_mytype |= Base.unwrap_unionall(mi.specTypes).parameters[2] === PC_A.MyType
-        end
-        have_mytype && @warn "Code in setup_workload block was precompiled"
-        # Check that calls inside @compile_workload are precompiled
-        m = only(methods(PC_A.call_findfirst))
-        count = 0
-        for mi in specializations(m)
-            mi === nothing && continue
-            sig = Base.unwrap_unionall(mi.specTypes)
-            if sig.parameters[2] == PC_A.MyType && sig.parameters[3] == Vector{PC_A.MyType}
-                count += 1
-            end
-        end
-        @test count == 1
-        # Even one that was runtime-dispatched
-        m = which(Tuple{typeof(findfirst), Base.Fix2{typeof(==), T}, Vector{T}} where T)
-        count = 0
-        for mi in specializations(m)
-            mi === nothing && continue
-            sig = Base.unwrap_unionall(mi.specTypes)
-            if sig.parameters[3] == Vector{PC_A.MyType}
-                count += 1
-            end
+    # Check that calls inside @setup_workload are not precompiled
+    m = which(Tuple{typeof(Base.vect), Vararg{T}} where T)
+    have_mytype = false
+    for mi in specializations(m)
+        mi === nothing && continue
+        have_mytype |= Base.unwrap_unionall(mi.specTypes).parameters[2] === PC_A.MyType
+    end
+    have_mytype && @warn "Code in setup_workload block was precompiled"
+    # Check that calls inside @compile_workload are precompiled
+    m = only(methods(PC_A.call_findfirst))
+    count = 0
+    for mi in specializations(m)
+        mi === nothing && continue
+        sig = Base.unwrap_unionall(mi.specTypes)
+        if sig.parameters[2] == PC_A.MyType && sig.parameters[3] == Vector{PC_A.MyType}
```

Comment:
It would probably be safer to specify this as a subtyping query instead of a structural query, since the system could in theory generate other specializations that you didn't expect to handle, and subtyping already has all of the length and structural queries to handle those
```
if sig == Tuple{typeof(PC_A.call_findfirst), PC_A.MyType, Vector{PC_A.MyType}}
```